### PR TITLE
Fce 646/prettier fishjam chat

### PR DESCRIPTION
**Pull request by czerwiukk**
## Description

Add prettier version of fishjam chat.

## Motivation and Context

We'd like to have a version that is slick enough to show it to the world.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to
      not work as expected)

### CODE REVIEW

File: examples/react-client/prettier-chat/src/components/RoomView.tsx [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+  const trackAmount = [localPeer, ...peers]
+    .filter(nonNullablePredicate)
+    .reduce((acc, curr) => {
+      if (curr.cameraTrack) acc++;
+      if (curr.screenShareVideoTrack) acc++;
+      return acc;
+    }, 0);
+
+  const shape = trackAmount === 2 ? 2 : Math.ceil(Math.sqrt(trackAmount));
```
Comment (kamil-stasiak) [inline examples/react-client/prettier-chat/src/components/RoomView.tsx:18]: nit: We could extract it

File: examples/react-client/prettier-chat/src/components/SettingsSheet.tsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+          <SheetDescription>
+            Adjust your camera, microphone, and other settings.
+          </SheetDescription>
+        </SheetHeader>
+
+        <section className="space-y-4 mt-4">
+          <div className="space-y-2">
+            <Label>Camera settings</Label>
+
+            <CameraSettings />
```
Comment (kamil-stasiak) [inline examples/react-client/prettier-chat/src/components/SettingsSheet.tsx:29]: If the user clicks on `Camera settings` if access to camera is blocked the whole site will crash. Mic also has this problem
image

File: examples/react-client/prettier-chat/src/components/RoomView.tsx
Comment (kamil-stasiak) [inline examples/react-client/prettier-chat/src/components/RoomView.tsx:1]: There is an error on this screen:

image

File: examples/react-client/prettier-chat/vite.config.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import { defineConfig } from "vite";
+import react from "@vitejs/plugin-react-swc";
+import path from "path";
+
+// https://vitejs.dev/config/
+export default defineConfig({
+  plugins: [react()],
```
Comment (kamil-stasiak) [inline examples/react-client/prettier-chat/vite.config.ts:7]: suggestion: I would set the default port to, for example, 3010 to make a clear distinction from the old app

**Review by kamil-stasiak — CHANGES_REQUESTED** [review]
Wow, nice app! Nice job! 

I left a few suggestions, but the only important ones are those with errors

File: examples/react-client/prettier-chat/src/lib/roomManager.ts [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+  room: BasicInfo;
+  peer: BasicInfo;
+};
+
+export const getRoomCredentials = async (
+  roomManagerUrl: string,
+  roomName: string,
+  peerName: string,
+) => {
+  const res = await axios.get<RoomManagerResponse>(roomManagerUrl, {
```
[the remark this answers is not in the document]
Comment (czerwiukk) [reply examples/react-client/prettier-chat/src/lib/roomManager.ts:16]: my approach is to promote good solutions, axios makes performing such api calls really simple

File: examples/react-client/prettier-chat/src/components/SettingsSheet.tsx (hunk 2 above)
Comment (czerwiukk) [reply examples/react-client/prettier-chat/src/components/SettingsSheet.tsx:29]: d

File: examples/react-client/prettier-chat/vite.config.ts (hunk 3 above)
Comment (czerwiukk) [reply examples/react-client/prettier-chat/vite.config.ts:7]: d

File: examples/react-client/prettier-chat/src/components/ui/switch.tsx
[the remark this answers is not in the document]
Comment (czerwiukk) [reply examples/react-client/prettier-chat/src/components/ui/switch.tsx:1]: It's already separated to it's isolated folder, but still kept in the codebase for possible adjustments. That's the philosophy behind [shadcn library](https://ui.shadcn.com/docs).

File: examples/react-client/prettier-chat/src/components/RoomView.tsx (hunk 1 above)
Comment (czerwiukk) [reply examples/react-client/prettier-chat/src/components/RoomView.tsx:18]: why should we?

File: examples/react-client/prettier-chat/src/hooks/useAutoConnect.ts [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+  const isPeerIdle = peerStatus === "idle";
+
+  const handleConnection = useCallback(async () => {
+    if (!isPeerIdle || !roomManagerUrl || !roomName || !peerName) return;
+    const { url, peerToken } = await getRoomCredentials(
+      roomManagerUrl,
+      roomName,
+      peerName,
+    );
+    await connect({ url, token: peerToken });
```
[the remark this answers is not in the document]
Comment (czerwiukk) [reply examples/react-client/prettier-chat/src/hooks/useAutoConnect.ts:28]: d

**Review by kamil-stasiak — APPROVED** [review]
Nice! 

We could add "Toggle Camera" and "Toggle Microphone" buttons to allow joining the room without any media streams

**Pull request merged**